### DISCUSSION TITLE
how to do this

### DISCUSSION THREAD

**Discussion by BenskuPro** [mega12345mega/NBT-Editor#43] [Q&A] [answered]
<img width="1440" alt="Снимок экрана 2023-04-22 в 13 04 44" src="https://user-images.githubusercontent.com/131523313/233777634-91992358-eab9-451b-b6da-23cb7506f656.png">
Im asking how to do this
Im not 100% sure that its made by this mod, but i somehow made it before. But i forgot how to do it

**Comment by mega12345mega** [comment, marked as answer]
There isn't currently a way to create lore images. It was suggested a long time ago, but I never ended up adding it.
I did some searching, and I was surprised that the best alternative that I could find was a very outdated Spigot plugin: [https://www.spigotmc.org/resources/image-reader-images-in-chat-item-lore.16077/](https://www.spigotmc.org/resources/image-reader-images-in-chat-item-lore.16077/).
I did find [https://cloudwolfyt.github.io/pages/gens/image-json.html](https://cloudwolfyt.github.io/pages/gens/image-json.html) which conveniently runs in the browser, but it doesn't seem to work for lore, only `/tellraw`. This isn't very easy to convert, since `/tellraw` allows `\n` and each additional entry doesn't go onto a new line.

For now I've written a quick and dirty JavaScript converter program. Copy and paste this code into the console of a new tab (Ctrl + Shift + I > Console), and then run `tellrawToItem('<copy the output of the website here>');` and then copy and paste the output into the game. (Make sure to enable the extended chat limit in `/nbt config`)
```
function tellrawToItem(tellraw) {
    var squares = JSON.parse(tellraw.substring("tellraw @s ".length).replaceAll("\n", "\\n"));
    var lines = [];
    var line = [];
    for (var square of squares) {
        if (square.text.startsWith("\n")) {
            lines.push(line);
            line = [{text: square.text.substring(1), color: square.color}];
        } else {
            line.push(square);
        }
    }
    lines.push(line);
    var output = [];
    for (var line of lines) {
        output.push(JSON.stringify({text: "", extra: line}));
    }
    document.body.textContent = "/get item painting{display:{Lore:" + JSON.stringify(output).replaceAll("\\\\", "\\") + "}}";
}
```

Since this has been suggested a couple times now, I'll prioritize it!

**Reply by BenskuPro** [reply]
ok, i just remember before when i played i had like 60 client-side mods. so its prob one of them

**Reply by BenskuPro** [reply]
btw, if i remember right it was possible to make images like that not only in names of item. It was possible to make a images like in game like name of mob. You needed to give link of image that u wanna make, then it gives u a shulker. U open shulker, there is armor stands. Then place dispenser, put the armor stands in dispenser (u need to set dispenser so that watches up). Then press button many thimes or do the with observer infinity redstone signall thing. Then it places all the armor stands. When all stands are placed the image is ready.

**Reply by mega12345mega** [reply]
I've finally released this feature in v1.12.0! Drag the image onto the inventory screen (like .nbt files) for a new item, drag onto the new display gui to append to the current lore, and drag onto the book editor to append to the current page.
For the more advanced features like spawning the image into the world, refer to the mod recommended by @HyperSoop in the issue I created for this: [https://modrinth.com/mod/fzmm](https://modrinth.com/mod/fzmm).
